tests/test_comparison_template_lib.py

```diff
@@ -0,0 +1,149 @@
+import pandas as pd
+import pytest
+
+import splink.duckdb.duckdb_comparison_template_library as ctld
+import splink.spark.spark_comparison_template_library as ctls
+from splink.duckdb.duckdb_linker import DuckDBLinker
+from splink.spark.spark_linker import SparkLinker
+
+
+def test_duckdb_date_comparison_run():
+    ctld.date_comparison("date")
+
+
+def test_spark_date_comparison_run():
+    ctls.date_comparison("date")
```

Comment:
`parametrize` these too? They're identical tests, but for the cll path.

Reply:
Good point

Reply:
have added a separate `parametrize` call for this not test. Not sure if there is a more efficient way of writing it given the last test is using the same params

Reply:
One option would be to leave it as it is currently (which I think is fine - not a huge amount of repetition), and have a separate PR for defining a parameterised test fixture which contains the linker and the libraries. This would then be used in tests something like:
```py
def test_some_splink_thing(dialect_fixture):
    cl = dialect_fixture.params["cl"]
    cll = dialect_fixture.params["cll"]
    Linker = dialect_fixture.params["linker"]
    # actual test logic etc.
```
It adds a bit more complexity to the overall test set-up (and potentially that bit of boilerplate to tests), but it _does_ mean we could easily test generic features against multiple backends, whereas I think there are certain things where atm we only cover `duckdb`.

Reply:
It also may get a bit complicated if certain tests only use certain backends (and then we have a bunch of fixtures for different combinations?), so may not be worth the overhead of it all